packages/workspace/src/browser/workspace-commands.ts

```diff
@@ -405,6 +405,19 @@ export class WorkspaceCommandContribution implements CommandContribution {
         }
     }
 
+    protected areMultipleOpenHandlersPresent(openers: OpenHandler[], uri: URI): boolean {
+        let count = 0;
+        for (const opener of openers) {
+            if (opener.canHandle(uri) > 0) {
```

Comment:
@akosyakov @adi7ya
I see an issue here. `canHandle` returns `MaybePromise<number>` and command `isVisible` returns a synced boolean. I don't really see how we can overcome this without changing the whole command interface as describe here: https://github.com/theia-ide/theia/issues/435 which is depend on our menus library that doesn't support async APIs... 

Can we revert this and make Open With always visible?

Reply:
@amiramw It would be good to file a proper issue first with the reproducible example. I don't think we should revert it, but we could improve it by checking for example whether a result is promised and if it is so treat it as visible and enabled for now.

Reply:
but then there will always be more than one `canHandle` that is either returning true (code editor) or promise (EditorPreviewManager for example).

Reply:
remove `@theia/editor-preview` extension for testing? for example Gitpod does not use it

Reply:
oh, so maybe I can implement what you suggested and if the specific deployment is lucky enough not to load any promise implementation of an opener he will benefit from this. 
also i would like to make it configurable. 

does it sound ok?

Reply:
I don't think it should be configurable, i would rather think how to handle promises properly. Please open an issue that others were aware about this conversation. Not many are checking comments on closed pull requests.